components/icon-button/index.js

```diff
@@ -22,7 +22,7 @@ import Dashicon from '../dashicon';
 class IconButton extends Component {
 	render() {
 		const { icon, children, label, className, tooltip, focus, ...additionalProps } = this.props;
-		const classes = classnames( 'components-icon-button', className );
+		const classes = classnames( { 'components-icon-button': icon }, className );
```

Comment:
An icon button without an icon is not really an icon button, it's just a button and we should just use Button in that case